src/filters.rs

```diff
@@ -435,4 +469,13 @@ mod tests {
         assert_eq!(unit!(split, tos!("a~b"), &[tos!("~")]),
                    Array(vec![tos!("a"), tos!("b")]));
     }
+
+    // unit! macro doesn't like an Array as an input so will
+    // skip unit test in favor of more complete test.
+    // Not sure how to adjust the macro to support Array...
+    // #[test]
+    // fn unit_join() {
+    //     assert_eq!(unit!(join, Array(vec![tos!("a"), tos!("b"),
+    //                tos!("c")]), &[tos!(",")], tos!("a,b,c"));
```

Comment:
you don't have to use `unit!`, you can simply call `join` normally and check the result. It's a bit more verbose, but does the job.